azurelinuxagent/common/protocol/wire.py

```diff
@@ -893,9 +880,23 @@ def get_vm_settings():
             else:  # since the vmSettings were updated, the response must include an etag
                 raise ProtocolError("The vmSettings do no include an Etag [correlation ID: {0}]".format(correlation_id))
 
-            logger.info("Fetched new vmSettings [correlation ID: {0} New eTag: {1}]", correlation_id, response_etag)
             response_content = self.decode_config(response.read())
-            return response_etag, response_content
+            vm_settings = ExtensionsGoalStateFactory.create_from_vm_settings(response_etag, response_content)
+
+            # log the HostGAPlugin version when we fetch the vmSettings for the first time
+            is_first_vm_settings = self._vm_settings_goal_state is None
+            if is_first_vm_settings:
+                message = "HostGAPlugin version: {0}".format(vm_settings.host_ga_plugin_version)
+                logger.info(message)
+                add_event(op=WALAEventOperation.HostPlugin, message=message, is_success=True)
```

Comment:
If the HGAP version < .115, we will keep emitting this event on every call to `update_goal_state`. We should only log this once irrespective

Reply:
Yes, good point. I have another change in the pipeline where if we do not support vmsettings we will stop trying to use it for several hours. I'll add the check for minimum version there, too.

Reply:
Cool cool. So this will be taken care in the next PR?

Reply:
Yes, I need some refactoring to get accurate telemetry and need to touch the error handling code first. That is the reason I postponed the handling of not supported. Version check is just a variation of not supported.